### PR TITLE
feat(tracing): Add native application start spans

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] New feature

## :scroll: Description

This PR adds native application start spans as children of the root app start span reported by RN today. 

## Example from iOS

Screenshot 2024-06-04 at 17 50 47

## Example from Android

Screenshot 2024-06-04 at 17 52 28

##  ✅  Waiting for native releases

- https://github.com/getsentry/sentry-cocoa/pull/4021
- https://github.com/getsentry/sentry-java/pull/3454

## :bulb: Motivation and Context

Doesn't close https://github.com/getsentry/sentry-react-native/issues/3445 as also RN specifics spans needs to be added. That will be separate PR.

- https://github.com/getsentry/sentry-react-native/issues/3445

## :green_heart: How did you test it?
sample app, unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps
- Release `sentry-cocoa` with https://github.com/getsentry/sentry-cocoa/pull/4021
- Release `sentry-android` with https://github.com/getsentry/sentry-java/pull/3454

### CODE REVIEW

File: ios/RNSentry.mm
Comment (lucas-zimerman) [inline ios/RNSentry.mm:382]: appStartMeasurementWithSpans is missing?

Comment (lucas-zimerman) [reply ios/RNSentry.mm:382]: I saw that you added it on https://github.com/getsentry/sentry-cocoa/pull/4021 but a new release was released before the inclusion of it.
We should wait for a new release of sentry cocoa before merging this PR

Comment (krystofwoldrich) [reply ios/RNSentry.mm:382]: Yes. Exactly. 👍

File: test/tracing/reactnativetracing.test.ts
Comment (lucas-zimerman) [inline test/tracing/reactnativetracing.test.ts:301]: https://github.com/getsentry/sentry-react-native/pull/3855/files#diff-8fe4b2c9966b386116cb1a00d65b34b257134d0f598817b95e4052546df00bf8R320-R323

This test should be updated to use `has_fetched` instead of `didFetchAppStart`

Comment (lucas-zimerman) [inline test/tracing/reactnativetracing.test.ts:301]: https://github.com/getsentry/sentry-react-native/pull/3855/files#diff-8fe4b2c9966b386116cb1a00d65b34b257134d0f598817b95e4052546df00bf8R483-R490

This test should be updated to use `has_fetched` instead of `didFetchAppStart`

Comment (lucas-zimerman) [reply test/tracing/reactnativetracing.test.ts:301]: This commend and the above one are working but I recommend to rename the test and the variable to avoid confusion in the future with a test  mentioning a variable that no longer exists.

**Review by lucas-zimerman — APPROVED** [review]
Thank you for the  PR! LGTM!

**Pull request merged**
